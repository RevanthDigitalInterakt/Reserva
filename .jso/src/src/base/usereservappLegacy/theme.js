  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.theme = undefined;
  var tokens = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  var theme = exports.theme = {
    space: {
      quarck: 4,
      nano: 8,
      micro: 12,
      xxxs: 16,
      xxs: 24,
      xs: 32,
      sm: 40,
      md: 48,
      xl: 64,
      xxl: 80
    },
    shadows: {
      bottomBarShadow: '0px -3px 2px rgba(0,0,0,0.1)',
      topBarShadow: '0px 3px 2px rgba(0,0,0,0.1)'
    },
    colors: {
      preto: tokens.preto,
      vermelhoFechadoRSV: tokens.vermelhoFechadoRSV,
      marinho: tokens.marinho,
      neutroFrio2: tokens.neutroFrio2,
      neutroQuente2: tokens.neutroQuente2,
      vermelhoRSV: tokens.vermelhoRSV,
      verde: tokens.verde,
      neutroQuente1: tokens.neutroQuente1,
      offWhite: tokens.offWhite,
      offWhite15: tokens.offWhite15,
      whiteSecondary: tokens.whiteSecondary,
      neutroFrio1: tokens.neutroFrio1,
      leve: tokens.leve,
      vermelhoAlerta: tokens.vermelhoAlerta,
      verdeSucesso: tokens.verdeSucesso,
      amareloAtencao: tokens.amareloAtencao,
      dropDownBorderColor: tokens.dropDownBorderColor,
      white: tokens.white,
      fullBlack: tokens.fullBlack,
      divider: tokens.divider,
      backgoundInput: tokens.backgoundInput,
      backgroundApp: tokens.backgroundApp,
      transparente: 'transparent',
      modalBackDropColor: tokens.modalBackDropColor,
      borderButton: tokens.borderButton,
      dividerCounter: tokens.dividerCounter,
      backgoundDivider: tokens.backgoundDivider,
      progressTextColor: tokens.progressTextColor,
      backgroundMenuOpened: tokens.backgroundMenuOpened,
      searchBarTextColor: tokens.searchBarTextColor,
      borderColorProduct: tokens.borderColorProduct
    },
    fonts: {
      nunitoBlack: 'Nunito-Black',
      nunitoBlackItalic: 'Nunito-BlackItalic',
      nunitoBold: 'Nunito-Bold',
      nunitoBoldItalic: 'Nunito-BoldItalic',
      nunitoExtraBold: 'Nunito-ExtraBold',
      nunitoExtraBoldItalic: 'Nunito-ExtraBoldItalic',
      nunitoExtraLight: 'Nunito-ExtraLight',
      nunitoExtraLightItalic: 'Nunito-ExtraLightItalic',
      nunitoItalic: 'Nunito-Italic',
      nunitoLight: 'Nunito-Light',
      nunitoLightItalic: 'Nunito-LightItalic',
      nunitoRegular: 'Nunito-Regular',
      nunitoSemiBold: 'Nunito-SemiBold',
      nunitoSemiBoldItalic: 'Nunito-SemiBoldItalic',
      reservaDisplayRegular: 'ReservaDisplay-Regular',
      reservaSerifBlack: 'ReservaSerif-Black',
      reservaSerifBlackItalic: 'ReservaSerif-BlackItalic',
      reservaSerifBold: 'ReservaSerif-Bold',
      reservaSerifBoldItalic: 'ReservaSerif-BoldItalic',
      reservaSerifItalic: 'ReservaSerif-Italic',
      reservaSerifLight: 'ReservaSerif-Light',
      reservaSerifLightItalic: 'ReservaSerif-LightItalic',
      reservaSerifMedium: 'ReservaSerif-Medium',
      reservaSerifMediumItalic: 'ReservaSerif-MediumItalic',
      reservaSerifRegular: 'ReservaSerif-Regular',
      reservaSerifThin: 'ReservaSerif-Thin',
      reservaSerifThinItalic: 'ReservaSerif-ThinItalic',
      reservaSansBlack: 'ReservaSans-Black',
      reservaSansBlackItalic: 'ReservaSans-BlackItalic',
      reservaSansBold: 'ReservaSans-Bold',
      reservaSansBoldItalic: 'ReservaSans-BoldItalic',
      reservaSansItalic: 'ReservaSans-Italic',
      reservaSansLight: 'ReservaSans-Light',
      reservaSansLightItalic: 'ReservaSans-LightItalic',
      reservaSansMedium: 'ReservaSans-Medium',
      reservaSansMediumItalic: 'ReservaSans-MediumItalic',
      reservaSansRegular: 'ReservaSans-Regular',
      reservaSansThin: 'ReservaSans-Thin',
      reservaSansThinItalic: 'ReservaSans-ThinItalic'
    },
    fontWeights: {
      light: 300,
      normal: 400,
      medium: 500,
      SemiBold: 600,
      bold: 700,
      black: 900
    },
    fontSizes: [4, 8, 12, 14, 16, 18, 20, 24, 27, 32, 48],
    radii: {
      pico: 2,
      nano: 5,
      xxxs: 10,
      xs: 15,
      sm: 20,
      pill: 500,
      infinity: 999999
    },
    borderWidths: {
      none: tokens.borderWidthNone,
      hairline: tokens.borderWidthHairline,
      thin: tokens.borderWidthThin,
      thick: tokens.borderWidthThick,
      heavy: tokens.borderWidthHeavy,
      borderDiscountTag: tokens.borderDiscountTag,
      stepper: 3
    }
  };
