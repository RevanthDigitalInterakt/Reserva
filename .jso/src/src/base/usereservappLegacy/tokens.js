  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.whiteSecondary = exports.white = exports.vermelhoRSV = exports.vermelhoFechadoRSV = exports.vermelhoAlerta = exports.verdeSucesso = exports.verde = exports.searchBarTextColor = exports.progressTextColor = exports.preto = exports.offWhite15 = exports.offWhite = exports.neutroQuente2 = exports.neutroQuente1 = exports.neutroFrio2 = exports.neutroFrio1 = exports.modalBackDropColor = exports.marinho = exports.leve = exports.fullBlack = exports.dropDownBorderColor = exports.dividerCounter = exports.divider = exports.borderWidthThin = exports.borderWidthThick = exports.borderWidthNone = exports.borderWidthHeavy = exports.borderWidthHairline = exports.borderDiscountTag = exports.borderColorProduct = exports.borderButton = exports.backgroundMenuOpened = exports.backgroundApp = exports.backgoundInput = exports.backgoundDivider = exports.amareloAtencao = undefined;
  var white = exports.white = '#FFF';
  var fullBlack = exports.fullBlack = '#000';

  // cores primarias
  var preto = exports.preto = '#333333';
  var vermelhoFechadoRSV = exports.vermelhoFechadoRSV = '#5A0000';
  var marinho = exports.marinho = '#13294B';
  var neutroFrio2 = exports.neutroFrio2 = '#8A8C8E';
  var neutroQuente2 = exports.neutroQuente2 = '#B1ACA8';
  var divider = exports.divider = '#D8D9DA';
  var dividerCounter = exports.dividerCounter = '#EAEAEA';
  var progressTextColor = exports.progressTextColor = '#979797';
  var searchBarTextColor = exports.searchBarTextColor = '#666666';

  // cores secundarias
  var vermelhoRSV = exports.vermelhoRSV = '#A00000';
  var verde = exports.verde = '#22372B';
  var neutroQuente1 = exports.neutroQuente1 = '#D7CFCA';
  var offWhite = exports.offWhite = '#D6D1C4';
  var offWhite15 = exports.offWhite15 = '#F9F8F6';
  var whiteSecondary = exports.whiteSecondary = '#FAFAFA';
  var neutroFrio1 = exports.neutroFrio1 = '#D8D9DA';
  var leve = exports.leve = '#F9E1DD';

  // cores do sistema
  var vermelhoAlerta = exports.vermelhoAlerta = '#EF1E1E';
  var verdeSucesso = exports.verdeSucesso = '#38A238';
  var amareloAtencao = exports.amareloAtencao = '#FEA528';
  var dropDownBorderColor = exports.dropDownBorderColor = '#DDDDDD';

  // larguras de borda
  var borderWidthNone = exports.borderWidthNone = 0;
  var borderWidthHairline = exports.borderWidthHairline = 1;
  var borderWidthThin = exports.borderWidthThin = 2;
  var borderWidthThick = exports.borderWidthThick = 4;
  var borderWidthHeavy = exports.borderWidthHeavy = 8;
  var borderDiscountTag = exports.borderDiscountTag = 12;

  // cores de borda
  var borderButton = exports.borderButton = '#333333';
  var borderColorProduct = exports.borderColorProduct = '#757575';

  // backgrounds
  var backgroundMenuOpened = exports.backgroundMenuOpened = '#F1F1F1';
  var backgroundApp = exports.backgroundApp = '#F9F9F9';
  var modalBackDropColor = exports.modalBackDropColor = '#AFAFAF';
  var backgoundInput = exports.backgoundInput = '#F0F0F0';
  var backgoundDivider = exports.backgoundDivider = '#F5F5F5';
