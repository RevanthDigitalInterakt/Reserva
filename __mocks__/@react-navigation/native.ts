const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
