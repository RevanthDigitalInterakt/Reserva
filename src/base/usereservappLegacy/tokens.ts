export const white = '#FFF';
export const fullBlack = '#000';

// cores primarias
export const preto = '#333333';
export const vermelhoFechadoRSV = '#5A0000';
export const marinho = '#13294B';
export const neutroFrio2 = '#8A8C8E';
export const neutroQuente2 = '#B1ACA8';
export const divider = '#D8D9DA';
export const dividerCounter = '#EAEAEA';
export const progressTextColor = '#979797';
export const searchBarTextColor = '#666666';

// cores secundarias
export const vermelhoRSV = '#A00000';
export const verde = '#22372B';
export const neutroQuente1 = '#D7CFCA';
export const offWhite = '#D6D1C4';
export const offWhite15 = '#F9F8F6';
export const whiteSecondary = '#FAFAFA';
export const neutroFrio1 = '#D8D9DA';
export const leve = '#F9E1DD';

// cores do sistema
export const vermelhoAlerta = '#EF1E1E';
export const verdeSucesso = '#38A238';
export const amareloAtencao = '#FEA528';
export const dropDownBorderColor = '#DDDDDD';

// larguras de borda
export const borderWidthNone = 0;
export const borderWidthHairline = 1;
export const borderWidthThin = 2;
export const borderWidthThick = 4;
export const borderWidthHeavy = 8;
export const borderDiscountTag = 12;

// cores de borda
export const borderButton = '#333333';
export const borderColorProduct = '#757575';

// backgrounds
export const backgroundMenuOpened = '#F1F1F1';
export const backgroundApp = '#F9F9F9';
export const modalBackDropColor = '#AFAFAF';
export const backgoundInput = '#F0F0F0';
export const backgoundDivider = '#F5F5F5';
