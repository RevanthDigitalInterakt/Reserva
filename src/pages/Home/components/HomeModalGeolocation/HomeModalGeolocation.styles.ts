import { StyleSheet } from 'react-native';
import { COLORS, FONTS } from '../../../../base/styles';

export const styles = StyleSheet.create({
  modalContainer: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: 'rgba(0, 0, 0, 0.6)',
  },

  modalContent: {
    alignSelf: 'stretch',
    flex: 1,
    backgroundColor: COLORS.WHITE,
    borderRadius: 20,
    top: 430,
    padding: 20,
    alignItems: 'center',
  },

  modalHeader: {
    alignSelf: 'stretch',
    alignItems: 'flex-end',
  },

  modalTitle: {
    fontFamily: FONTS.NUNITO_BOLD,
    fontSize: 17,
  },

  modalParagraph: {
    fontFamily: FONTS.NUNITO_REGULAR,
    fontSize: 15,
    textAlign: 'center',
    marginVertical: 10,
  },

  modalPickerContainer: {
    alignSelf: 'stretch',
    marginVertical: 10,
    borderWidth: 1,
    borderColor: COLORS.LIGHT_GRAY,
    borderRadius: 8,
    padding: 15,
  },

  modalPickerContent: {
    fontFamily: FONTS.NUNITO_REGULAR,
    fontSize: 20,
    color: COLORS.LIGHT_GRAY,
  },

  modalActionButton: {
    backgroundColor: COLORS.BLACK,
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'stretch',
    marginVertical: 10,
    borderRadius: 50,
    padding: 20,
  },

  modalActionButtonText: {
    color: COLORS.WHITE,
    fontFamily: FONTS.NUNITO_SEMI_BOLD,
  },
});
