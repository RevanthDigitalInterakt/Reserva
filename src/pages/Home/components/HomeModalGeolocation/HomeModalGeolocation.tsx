import React, { useCallback, useEffect, useState } from 'react';
import {
  Modal,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { styles } from './HomeModalGeolocation.styles';
import useModalGeolocationStore from '../../../../zustand/useModalGeolocationStore';
import IconClose from '../../../../../assets/icons/IconClose';
import IconGeolocation from '../../../../../assets/icons/IconGeolocation';
import { Picker } from '../../../../components/Picker/Picker';
import { useHomeStore } from '../../../../zustand/useHomeStore';
import useAsyncStorageProvider from '../../../../hooks/useAsyncStorageProvider';
import { COLORS } from '../../../../base/styles';
import EventProvider from '../../../../utils/EventProvider';

export default function HomeModalGeolocation() {
  const { showModalGeolocation, modalGeolocationController } = useModalGeolocationStore(['showModalGeolocation', 'modalGeolocationController']);
  const { onSelectStateGeolocation, selectedStateGeolocation } = useHomeStore(['onSelectStateGeolocation', 'selectedStateGeolocation']);
  const { setItem } = useAsyncStorageProvider();

  const [visiblePicker, setVisiblePicker] = useState(false);
  const [selectedState, setSelectedState] = useState('');

  const statesAcronym = [
    {
      text: 'AC',
      subText: '',
    },
    {
      text: 'AL',
      subText: '',
    },
    {
      text: 'AP',
      subText: '',
    },
    {
      text: 'AM',
      subText: '',
    },
    {
      text: 'BA',
      subText: '',
    },
    {
      text: 'CE',
      subText: '',
    },
    {
      text: 'DF',
      subText: '',
    },
    {
      text: 'ES',
      subText: '',
    },
    {
      text: 'GO',
      subText: '',
    },
    {
      text: 'MA',
      subText: '',
    },
    {
      text: 'MT',
      subText: '',
    },
    {
      text: 'MS',
      subText: '',
    },
    {
      text: 'MG',
      subText: '',
    },
    {
      text: 'PA',
      subText: '',
    },
    {
      text: 'PB',
      subText: '',
    },
    {
      text: 'PR',
      subText: '',
    },
    {
      text: 'PE',
      subText: '',
    },
    {
      text: 'PI',
      subText: '',
    },
    {
      text: 'RJ',
      subText: '',
    },
    {
      text: 'RN',
      subText: '',
    },
    {
      text: 'RS',
      subText: '',
    },
    {
      text: 'RO',
      subText: '',
    },
    {
      text: 'RR',
      subText: '',
    },
    {
      text: 'SC',
      subText: '',
    },
    {
      text: 'SP',
      subText: '',
    },
    {
      text: 'SE',
      subText: '',
    },
    {
      text: 'TO',
      subText: '',
    },
  ];

  const saveUserGeolocation = useCallback(async () => {
    await setItem('User:Geolocation', selectedState);
    onSelectStateGeolocation(selectedState);
    modalGeolocationController(false);
  }, [selectedState]);

  const onCloseModal = useCallback(() => {
    modalGeolocationController(false);
    EventProvider.logEvent('modal_geolocation_close');
  }, []);

  const onSelectGeolocation = useCallback((geoState: string) => {
    setSelectedState(geoState);
    EventProvider.logEvent('banner_location', { banner_location: geoState });
  }, []);

  useEffect(() => {
    setSelectedState(selectedStateGeolocation);
  }, [selectedStateGeolocation]);

  return (
    <Modal animationType="fade" transparent visible={showModalGeolocation}>
      <View style={styles.modalContainer}>
        <View style={styles.modalContent}>
          <View style={styles.modalHeader}>
            <TouchableOpacity onPress={onCloseModal}>
              <IconClose />
            </TouchableOpacity>
          </View>
          <IconGeolocation />
          <Text style={styles.modalTitle}>
            Qual a sua localização?
          </Text>
          <Text style={styles.modalParagraph}>
            Compartilhe com a gente de onde você acessa a
            Reserva e tenha acesso a promoções exclusivas,
            fretes mais baratos e entregas mais rápidas
            quando disponível na sua região!
          </Text>

          <View style={styles.modalPickerContainer}>
            <TouchableOpacity onPress={() => setVisiblePicker(true)}>
              <Text style={styles.modalPickerContent}>
                {selectedState}
              </Text>
            </TouchableOpacity>
          </View>
          <Picker
            isVisible={visiblePicker}
            swipeDirection={false}
            onBackDropPress={() => {}}
            onAndroidBackButtonPress={() => { }}
            onClose={() => setVisiblePicker(false)}
            onSelect={(selected) => onSelectGeolocation(selected.text)}
            title="Selecione o Estado"
            items={statesAcronym}
          />
          <TouchableOpacity
            disabled={selectedState === 'Clique aqui para selecionar seu estado...'}
            onPress={saveUserGeolocation}
            style={[styles.modalActionButton,
              { backgroundColor: selectedState === 'Clique aqui para selecionar seu estado...' ? COLORS.LIGHT_GRAY : COLORS.BLACK },
            ]}
          >
            <Text style={styles.modalActionButtonText}>
              Salvar minha localização
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </Modal>
  );
}
