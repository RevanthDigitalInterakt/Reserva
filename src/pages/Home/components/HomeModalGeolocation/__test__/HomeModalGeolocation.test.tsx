import React from 'react';
import {
  render,
  screen,
} from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '../../../../../base/usereservappLegacy/theme';
import HomeModalGeolocation from '../HomeModalGeolocation';

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <HomeModalGeolocation />
  </ThemeProvider>
);

it('should match to snapshot', () => {
  render(TestingComponent);
  expect(screen.toJSON()).toMatchSnapshot();
});
