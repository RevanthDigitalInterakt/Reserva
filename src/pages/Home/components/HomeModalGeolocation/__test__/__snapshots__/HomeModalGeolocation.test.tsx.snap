// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match to snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
/>
`;
